src/pykeen/training/callbacks.py

```diff
@@ -190,15 +279,38 @@ class MultiTrainingCallback(TrainingCallback):
     #: A collection of callbacks
     callbacks: List[TrainingCallback]
 
-    def __init__(self, callbacks: TrainingCallbackHint = None) -> None:
-        """Initialize the callback."""
+    def __init__(
+        self,
+        callbacks: TrainingCallbackHint = None,
+        callback_kwargs: TrainingCallbackKwargsHint = None,
+    ) -> None:
+        """
+        Initialize the callback.
+
+        .. note ::
+            the constructor allows "broadcasting" of callbacks, i.e., proving a single callback,
+            but a list of callback kwargs. In this case, for each element of this list the given
+            callback is instantiated.
+
+        :param callbacks:
+            the callbacks
+        :param callback_kwargs:
+            additional keyword-based parameters for instantiating the callbacks
+        """
         super().__init__()
         if callbacks is None:
-            self.callbacks = []
-        elif isinstance(callbacks, TrainingCallback):
-            self.callbacks = [callbacks]
+            assert callback_kwargs is None
+            callbacks = []
+            callback_kwargs = []
         else:
-            self.callbacks = list(callbacks)
+            callbacks = upgrade_to_sequence(callbacks)
+            callback_kwargs = upgrade_to_sequence(callback_kwargs)
+        # callback broadcasting
+        if len(callback_kwargs) > 1 and len(callbacks) == 1:
```

Comment:
isn't this logic implemented somewhere else? can it be reused?

Reply:
no that I was aware of.

I moved it to `pykeen.utils` in 5db7c60